tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-table/attributelist-table.component.ts

```diff
@@ -215,6 +245,110 @@ export class AttributelistTableComponent implements AttributelistTable, OnInit,
     this.updateTable();
   }
 
+  /**
+   * Fired when a column filter is clicked.
+   */
+  public onFilterClick(columnName: string): void {
+    // Get the unique values for this column
+    this.valueParams.applicationLayer = this.dataSource.params.layerId;
+    this.valueParams.attributes = [];
+    this.valueParams.attributes.push (columnName);
+    this.valueService.uniqueValues(this.valueParams).subscribe((data: UniqueValuesResponse) => {
+      if (data.success) {
+        let uniqueValues: FilterValueSettings[];
+        uniqueValues = [];
+        const colObject = this.layerFilterValues.columns.find(c => c.name === columnName);
+        const colIndex = this.layerFilterValues.columns.findIndex(obj => obj.name === columnName);
+        if (colObject.uniqueValues.length === 0) {
+
+          data.uniqueValues[columnName].forEach(val => {
+            let filterValueSettings: FilterValueSettings;
+            filterValueSettings = {value: val, select: true};
+            uniqueValues.push(filterValueSettings);
+          })
+        } else {
+            colObject.uniqueValues.forEach(val => uniqueValues.push(Object.assign({}, val)))
+        }
+
+        const config = new MatDialogConfig();
+        config.data = {
+          colName: columnName,
+          values: uniqueValues,
+        };
+        const dialogRef = this.dialog.open(AttributelistFilterValuesFormComponent, config);
+        dialogRef.afterClosed().subscribe(filterSetting => {
+          // Do the filtering
+          if (filterSetting !== 'CANCEL') {
+            if (filterSetting === 'ON') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = config.data.values;
+              this.layerFilterValues.columns[colIndex].nullValue = false;
+              this.layerFilterValues.columns[colIndex].status = true;
+            } else if (filterSetting === 'NONE') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = config.data.values;
+              this.layerFilterValues.columns[colIndex].nullValue = true;
+              this.layerFilterValues.columns[colIndex].status = true;
+            } else if (filterSetting === 'OFF') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = [];
+              this.layerFilterValues.columns[colIndex].nullValue = false;
+              this.layerFilterValues.columns[colIndex].status = false;
+            }
+            this.createFilter();
+            this.updateTable();
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Create the CQL filter string
+   */
+  public createFilter(): void {
```

Comment:
De class wordt wel vrij groot. Wellicht het bouwen van het CQL filter in een helper-class zetten. Dit zal nog wel vaker gedaan moeten worden, dus goed om nu alvast generiek te maken

Reply:
Zal ik doen.

Reply:
Het maken van het CQL filter heb ik in attributelist-common/attributelist-filter geplaatst.
Was bijna van plan om er nog meer in te zetten (bijv. het zetten van de filter values), maar dat maakte het eerst even wat te complex. Is later misschien nog wel een optie.